yumi:
  hiqp_kinematics_controller:
    hiqp_preload_jnt_limits:
      - link_frame: yumi_link_1_r
        limitations: [1.4835, -2.94, 2.94] #dq_max, lower joint limit, upper joint limit

      - link_frame: yumi_link_2_r
        limitations: [1.4835, -2.5, 0.75] 

      - link_frame: yumi_link_3_r
        limitations: [1.7453, -2.94, 2.94]

      - link_frame: yumi_link_4_r
        limitations: [1.3090, -2.16, 1.4]

      - link_frame: yumi_link_5_r
        limitations: [2.2689, -5.07, 5.07] 

      - link_frame: yumi_link_6_r
        limitations: [2.2689, -1.54, 2.41]

      - link_frame: yumi_link_7_r
        limitations: [2.2689, -4.0, 4.0] 

    # hiqp_preload_geometric_primitives:
    #   - name: floor_avoidance_plane
    #     type: plane 
    #     frame_id: world
    #     visible: true
    #     color: [0.2, 0.2, 1.0, 0.5]
    #     parameters: [0.0, 0.0, 1.0, 0.115]

    #   - name: yumi_link_5_r_avoidance_sphere
    #     type: sphere 
    #     frame_id: yumi_link_5_r
    #     visible: true
    #     color: [1.0, 0.0, 0.0, 0.6]
    #     parameters: [-0.023, -0.04, 0.1, 0.04]

    # hiqp_preload_tasks:
    #   - name: mytask
    #     type: TaskGeometricProjection
    #     behaviour: ['DynamicsFirstOrder', '10']
    #     priority: 2
    #     visibility: 0
    #     parameters: ['point', 'point', 'mypoint1 = mypoint2']